drivers/rail/rail_netdev.c

```diff
@@ -0,0 +1,496 @@
+                                                                power_level_tx
+                                                                );
+            *((uint16_t *)val) = (uint16_t) power_tx_ddBm / 10;
+            ret = sizeof(uint16_t);
+            break;
+        case (NETOPT_RETRANS):      // todo
+            break;
+        case (NETOPT_PROMISCUOUSMODE):
+            if (dev->promiscuousMode == true) {
+                *((netopt_enable_t *)val) = NETOPT_ENABLE;
+            }
+            else {
+                *((netopt_enable_t *)val) = NETOPT_DISABLE;
+            }
+            ret = sizeof(netopt_enable_t);
+            break;
+        case (NETOPT_AUTOACK):
+            if (RAIL_IsAutoAckEnabled(dev->rhandle) == true) {
+                *((netopt_enable_t *)val) =  NETOPT_ENABLE;
+            }
+            else {
+                *((netopt_enable_t *)val) =  NETOPT_DISABLE;
+            }
+            ret = sizeof(netopt_enable_t);
+            break;
+        case (NETOPT_PRELOADING):   // todo what does it do??
+            break;
+        case (NETOPT_RAWMODE):
+            break;
+        case (NETOPT_CSMA):         // todo can it be switched on / off?
+            *((netopt_enable_t *)val) = NETOPT_ENABLE;
+            ret = sizeof(netopt_enable_t);
+            break;
+        case (NETOPT_CSMA_RETRIES):
+            assert(max_len >= sizeof(int8_t));
+            *((uint8_t *)val) = csma_config.csmaTries;
+            ret = sizeof(uint8_t);
+            break;
+        case (NETOPT_AUTOCCA):
+            break;
+        case (NETOPT_MAC_NO_SLEEP):
+            if (dev->macNoSleep == false) {
+                *((netopt_enable_t *)val) =  NETOPT_DISABLE;
+            }
+            else {
+                *((netopt_enable_t *)val) =  NETOPT_ENABLE;
+            }
+            ret = sizeof(netopt_enable_t);
+            break;
+        case (NETOPT_CHANNEL_HOP):
+            break;
+        case (NETOPT_CODING_RATE):
+            break;
+
+        case (NETOPT_BANDWIDTH):
+            /*
+               assert(max_len >= sizeof(int8_t));
+               // bits/seconds
+               uint32_t bw = RAIL_BitRateGet();
+               uint8_t bw_kb = bw/1000;
+             *((uint8_t *)val) = bw_kb;
+               ret = sizeof(uint8_t);
+             */
+            break;
+        case (NETOPT_CHANNEL_FREQUENCY):
+            break;
+        case (NETOPT_SINGLE_RECEIVE):
+            break;
+        case (NETOPT_IQ_INVERT):
+            break;
+        case (NETOPT_SRC_LEN):
+            break;
+        case (NETOPT_ADDRESS_LONG):
+            break;
+        case (NETOPT_DEVICE_TYPE):
+            break;
+        case (NETOPT_ADDRESS):
+            break;
+        case (NETOPT_CHANNEL):
+            break;
+        case (NETOPT_NID):
+            break;
+        case (NETOPT_SPREADING_FACTOR):
+            break;
+        case (NETOPT_ACK_REQ):
+            break;
+        case (NETOPT_STATS):
+            break;
```

Comment:
You're not forced to implement all the NETOPT, just remove the ones are not necessary.

Reply:
Yep. I did it to have a "list" of options, I might have to implement ;)
I tidy it up and move the list to a note.